testfm/helpers.py

```diff
@@ -14,13 +14,14 @@ def product():
 
 def run(command):
     """Use this helper to execute shell command on Satellite"""
-    return os.popen(f"ansible server -i testfm/inventory -u root -m command -a '{command}'").read()
+    return os.popen(f"ansible server -i testfm/inventory -u root -m shell -a '{command}'").read()
 
 
 def server():
     """Use this to find whether server on which tests are running is capsule or satellite."""
-    result = run("rpm -q satellite")
-    if "rc=0" in result:
-        return "satellite"
-    else:
-        return "capsule"
+    return "satellite" if "rc=0" in run("rpm -q satellite") else "capsule"
+
+
+def rhel7():
```

Comment:
@ogajduse I really appreciate your input and reviews here! will ask @shubhamsg199 to look into above if similar to robottelo can be implemented here.

As 7.0 is the last and only release based on both el7 + el8 and 7.1 will be only on el8 so we preferred to have a single method for `el7` herein 7.0 to adapt new config and files change wrt el8 in tests (can be removed in future branches).
@shubhamsg199 and I discussed this earlier after looking at similar upstream f-m implementation,
https://github.com/theforeman/foreman_maintain/blob/master/definitions/features/redis.rb#L22-L28

Reply:
@Gauravtalreja1 The example from foreman_maintain is most likely the best way to get the OS version.
However, I will leave the final implementation up to you both. Feel free to resolve this thread.

Reply:
Thanks for your input @ogajduse. Since testfm is based on ansible so we are using ansible ad-hoc to run the shell commands defined in the helpers.py 
The output of the ad-hoc commands is a `string`
https://github.com/SatelliteQE/testfm/blob/fe4c4f1a4588b1302c322b11388841f4854163a9/testfm/helpers.py#L15
Based on the robottelo implementation you recommended above, It requires `stdout` which is not possible here since the output is a string.
Also as @Gauravtalreja1 said 7.0 is the last release on both el7 and el8 so we decided to have a method only for el7

Reply:
You mentioned that testfm is Ansible-based. If you will use the following command with the respective environmental variables, you can get a pure JSON output that you can parse using Python.
```
ANSIBLE_LOAD_CALLBACK_PLUGINS=true ANSIBLE_SHOW_CUSTOM_STATS=0 ANSIBLE_STDOUT_CALLBACK=json ansible all -i <your-system>, -m setup
```

From the output, you should be able to get `ansible_distribution_major_version` variable which provides the major version.

As I said, I will leave the final implementation up to you. These are just my two cents.

Reply:
Appreciate your thoughts on it, I tried  this approach already and as I said the helper func we use gives us `string` output(even json enclosed in a string). It is possible to parse it but will make code complicated.
Hence will be using `rpm -qa rubygem-foreman_maintain | grep el7` as it would also be an additional check for foreman_maintain package.

Reply:
Resolving this thread as this is non-blocking atm, @ogajduse I really appreciate your review comments here and would suggest, feel free to open a PR if you find any good alternative for this.

Thanks!